README.md

## Overview
This is a mod for Krastorio2 that is meant to be played with low or no resources. Blood acts as a universal resource that can be processed into any basic resource, similar to the late-game Matter.

This mod does not make any changes to resource generation or biters difficulty, so it's up to the player to create a challenge. This mod also does not remove mining equipment.
Resource Spawner Overhaul (RSO) is recommended as it can be configured to generate very sparse resources, and also has an option for no starting resources. (Why do we always conveniently crash next to ore patches anyway?)
If playing without starting resources, then the early game may feel slow. There's only a pistol and 10 ammunition rounds to get the ball rolling. To speed things up, some "fast start" mods could be installed. Also, Krastorio has a 'bonus start items' option, and this mod also has a "blood richness multiplier" setting. Note you may have to lower difficulty of biters a bit (e.g. reduce time-based evolution by half or more) because getting resources won't be as easy as plopping down some miners.

Another challenge option is to use the Island map that has some resources to get the base started but would force player to transition to blood recipes at some point.

The most difficult resource to automate in this mod is Crude Oil, but there should be plenty of it once automated. Just like Krastorio2 forces player to go and kill some nests to automate Biomass production, this mod will require player to kill some nests to collect enough spawner eggs to research and automate Crude Oil production. The recipe is fairly generous, but still exercise caution if increasing technology cost multiplier.

## Loot Pickup
Blood crystals dropped by biters need to be picked up. Too many items lying on the ground will eventually slow down the game. This mod does not include any built-in option for loot pickup.

Here are some loot pickup mods that were tested:
- [Auto Deconstruct Loot, Reconstructed](https://mods.factorio.com/mod/auto_deconstruct_loot_reconstructed) by Derringer - works great for when construction robots are unlocked
- [Schall Pickup Tower](https://mods.factorio.com/mod/SchallPickupTower) by Schallfalke - reportedly will have UPS issues if a lot of them are placed, but with the ability to create biter farms not that many towers may be required
- [Loot Chest UPS](https://mods.factorio.com/mod/LootChestUPS) by ptx0 - should be very UPS-friendly, but could feel a bit too "magical" to some

Alternatively, you can get creative with belts picking up loot. They are a solid early game option.


## Blood, Wood, and Matter
This mod by default changes wood recipes so that they require a blood product. The goal here is to remove the "infinite matter with zero input" loop in Krastorio2. Additionally, the goal is for blood recipes to not become obsolete once matter recipes are unlocked. The idea is that matter recipes could be used for unwanted blood byproducts or for any excess of a resource. If you don't want to have wood recipes changed, there is a setting that disables the changes.

## Biter Farms
The mod allows placement of two tiers of biter spawners for blood farming. There's a caveat though. Spawners generate two kinds of biters: the "guardians" which stay close to the spawner and don't require pollution to spawn, and the "group attack" biters which spawn from absorbed pollution. The "guardians" attack on proximity, and they are the main source of blood in the early game if playing without starting resources. These "guardians", however, become "frozen" once player moves too far away from the spawner. The game freezes them to conserve resources. The pollution-induced "group attack" biters don't have this limitation. Design farms accordingly.


## Compatibility
This mod loads with Space Exploration (SE) but does not add any recipes for SE-specific resources. Additionally, Imersite won't be available from blood recipes if SE is installed because Imersite is not meant to be found on Nauvius with SE.

All resources from BZ mods by brevven are added to blood recipes if installed (Aluminum, Lead, Flake Graphite, Wolframite, Zircon, Titanium).

Armoured/Cold/Explosive biters drop twice the amount of blood crystals because of the extra hassle dealing with them. Their spawners also drop quite a bit more spawner eggs.

## Mods that may go well with Rich Blood
[Rampant Evolution](https://mods.factorio.com/mod/RampantEvolution) by Veden. This mod allows fine-tuning of what affects evolution. For example, it can reduce evolution by a flat amount for each killed biter. This would slow down evolution in the early game, but will be negligible in the late game because evolution reduction on kill doesn't scale with biter size. Rampant Evolution also allows for control of evolution using pollution. For example, pollution absorption by spawners could be reducing evolution, and pollution absorption by terrain and trees could be increasing evolution. Combined with biter farms, this can incentivise player to pave the terrain and place air filters strategically so that most pollution is absorbed by spawners.


## Localization
- [Aquilo](https://mods.factorio.com/user/Aquilo) (ru)

This mod uses modified assets of Krastorio2.


changelog.txt
---------------------------------------------------------------------------------------------------
Version: 0.1.7
Date: 2022-07-10
  Changes:
    - Add support for brevven's Tin mod.
---------------------------------------------------------------------------------------------------
Version: 0.1.6
Date: 2022-05-30
  Changes:
    - EN: This update requires MANUAL INTERVENTION. If you are using a wood recipe from Electrolyte, you need to flip input (R) on your greenhouses after this update.
    - RU: ВНИМАНИЕ! Если используйте рецепт древесины из Электролита, пожалуйста поменяйте ввод жидкостей в теплицах (R) после этого обновления.
    - Add a setting to not modify wood recipes. You can use this setting if you want to add the mod to an existing factory and do not want to redo wood production. Or use this setting if you just prefer to not have wood recipes changed.
    - Wood from Electrolyte recipe now always takes 30 seconds and requires 200 Water (previously AAI Industry was changing the values).
---------------------------------------------------------------------------------------------------
Version: 0.1.5
Date: 2022-05-18
  Changes:
    - Add Russian localization, thanks to Aquilo.
---------------------------------------------------------------------------------------------------
Version: 0.1.4
Date: 2022-05-15
  Changes:
    - Rework crude oil production loop to better balance the ratio of required machines and make things more OCD-friendly. Specific changes below.
    - Blood Separation: Increase crafting time from 8s to 60s. Produces 5 Blood Tissue instead of 20.
    - Biomass from Blood Tissue: Increase crafting time from 20s to 30s. Requires 5 Wood instead of 10. Requires 50 Oxygen instead of 100. Produces 5 Biomass instead of 10.
    - Stem cells from lab-grown embryo: Increase crafting time from 8s to 30s. Now has 80% chance to return Lab-grown embryo and 20% chance to return Empty lab capsule (no more Empty lab capsule sink). Due to how the game's % result works, the recipe will sometimes return nothing or will return both embryo and empty capsule. Not ideal, but over time the 80%-20% ratio will be true.
    - Stem cells from spawner egg: Increase crafting time from 8s to 30s.
    - Lab-grown embryo: Requires 5 Biomass instead of 10. Requires 5 Stem cell flasks instead of 10.
    - Lab-grown embryo liquefaction: Reduce crafting time from 15 to 12. No longer requires 5 Biomass.
    - Spawner egg liquefaction: Reduce crafting time from 30 to 24. No longer requires 5 Biomass.
---------------------------------------------------------------------------------------------------
Version: 0.1.3
Date: 2022-05-09
  Changes:
    - Fix BZ mods requirements.
---------------------------------------------------------------------------------------------------
Version: 0.1.2
Date: 2022-05-09
  Changes:
    - Fix not being able to add productivity modules to BZ ores catalytic electrolysis.
---------------------------------------------------------------------------------------------------
Version: 0.1.1
Date: 2022-05-05
  Changes:
    - Add Stone and Coal to Big blood crystal smelting.
    - Rebalance vanilla and BZ ores chances from blood crystals smelting.
    - Rebalance blood costs of some Catalytic electrolysis recipes.
    - Increase Dirty Water byproduct of Catalytic electrolysis recipes.
    - Blood crystal powder liquefaction now yields more blood per BZ mod installed. This is to prevent blood crystals smelting from becoming far better choice compared to blood recipes when BZ mods are installed.
    - Reduce Blood crystal powder cost in Wood recipe from 5 to 4.
    - Increase Nitrogen cost from 10 to 50 for Blood plasma barrel.
    - Fix Catalytic electrolysis for Uranium ore giving way too much ore.
---------------------------------------------------------------------------------------------------
Version: 0.1.0
Date: 2022-03-26
  Features:
    - Starting alpha version


prototypes/item.lua
data:extend(
{
  {
    type = "item-subgroup",
    name = "blood-processing",
    group = "intermediate-products",
    order = "a[blood]-1"
  },

  {
    type = "item-subgroup",
    name = "blood-resources",
    group = "intermediate-products",
    order = "a[blood]-2"
  }
})

data:extend(
{
  {
    type = "item",
    name = "blood-crystal-small",
    icon = "__RichBlood__/graphics/icons/blood-crystal-small.png",
    icon_size = 64,
    subgroup = "blood-processing",
    order = "a[blood]-10",
    stack_size = 200,
  },

  {
    type = "item",
    name = "blood-crystal-medium",
    icon = "__RichBlood__/graphics/icons/blood-crystal-medium.png",
    icon_size = 64,
    subgroup = "blood-processing",
    order = "a[blood]-15",
    stack_size = 200,
  },

  {
    type = "item",
    name = "blood-crystal-big",
    icon = "__RichBlood__/graphics/icons/blood-crystal-big.png",
    icon_size = 64,
    subgroup = "blood-processing",
    order = "a[blood]-20",
    stack_size = 200,
  },

  {
    type = "item",
    name = "spawner-egg",
    icon = "__RichBlood__/graphics/icons/spawner-egg.png",
    icon_size = 64,
    subgroup = "blood-processing",
    order = "a[blood]-25",
    stack_size = 200,
  },

  {
    type = "item",
    name = "lab-grown-embryo",
    icon = "__RichBlood__/graphics/icons/lab-grown-embryo.png",
    icon_size = 64,
    subgroup = "blood-processing",
    order = "a[blood]-30",
    stack_size = 50,
    burnt_result = "empty-lab-capsule",
  },

  {
    type = "item",
    name = "blood-crystal-powder",
    icon = "__RichBlood__/graphics/icons/blood-crystal-powder.png",
    icon_size = 64,
    subgroup = "blood-processing",
    order = "a[blood]-35",
    stack_size = 200,
  },

  {
    type = "fluid",
    name = "blood",
    default_temperature = 30,
    max_temperature = 40,
    auto_barrel = true,
    base_color = { r = 0.31, g = 0.18, b = 0.35 },
    flow_color = { r = 0.31, g = 0.18, b = 0.35 },
    icon = "__RichBlood__/graphics/icons/blood.png",
    icon_size = 64,
    icon_mipmaps = 4,
    subgroup = "blood-processing",
    order = "a[blood]-40",
  },

  {
    type = "fluid",
    name = "electrolyte",
    default_temperature = 30,
    max_temperature = 40,
    auto_barrel = true,
    base_color = { r = 0.83, g = 0.9, b = 0.0 },
    flow_color = { r = 1, g = 1, b = 1 },
    icon = "__RichBlood__/graphics/icons/electrolyte.png",
    icon_size = 64,
    icon_mipmaps = 4,
    subgroup = "blood-processing",
    order = "a[blood]-41",
  },

  {
    type = "item",
    name = "empty-cryogenic-flask",
    icon = "__RichBlood__/graphics/icons/empty-cryogenic-flask.png",
    icon_size = 64,
    subgroup = "intermediate-product",
    order = "a[blood]-45",
    stack_size = 50,
  },

  {
    type = "item",
    name = "empty-cryogenic-barrel",
    icon = "__RichBlood__/graphics/icons/empty-cryogenic-barrel.png",
    icon_size = 64,
    subgroup = "intermediate-product",
    order = "a[blood]-47",
    stack_size = 10,
  },

  {
    type = "item",
    name = "blood-plasma-barrel",
    icons =
   {
      {
        icon = "__RichBlood__/graphics/icons/empty-cryogenic-barrel.png",
        icon_size = 64
      },
      {
        icon = "__RichBlood__/graphics/icons/blood-plasma.png",
        icon_size =  64,
        scale = 0.22,
        shift = {0, 7}
      }
    },
    icon_size = 64,
    subgroup = "intermediate-product",
    order = "a[blood]-48",
    stack_size = 10,
  },
  
  {
    type = "item",
    name = "empty-lab-capsule",
    icon = "__RichBlood__/graphics/icons/empty-lab-capsule.png",
    icon_size = 64,
    subgroup = "intermediate-product",
    order = "a[blood]-50",
    stack_size = 50,
  },


  {
    type = "item",
    name = "stem-cells",
    icons =
   {
      {
        icon = "__RichBlood__/graphics/icons/empty-cryogenic-flask.png",
        icon_size = 64
      },
      {
        icon = "__RichBlood__/graphics/icons/stem-cells.png",
        icon_size =  64,
        scale = 0.25,
        shift = {0, 7}
      }
    },
    icon_size = 64,
    subgroup = "blood-processing",
    order = "a[blood]-55",
    stack_size = 50,
    burnt_result = "empty-cryogenic-flask",
  },
  {
    type = "item",
    name = "blood-tissue",
    icon = "__RichBlood__/graphics/icons/blood-tissue.png",
    icon_size = 64,
    subgroup = "blood-processing",
    order = "a[blood]-52",
    stack_size = 200,
  },
  {
    type = "item",
    name = "blood-centrifuge",
    icon = "__RichBlood__/graphics/icons/blood-centrifuge.png",
    icon_size = 64, icon_mipmaps = 1,
    flags = data.raw.item["centrifuge"].flags,
    subgroup = data.raw.item["centrifuge"].subgroup,
    order = data.raw.item["centrifuge"].order .. "-a",
    stack_size = data.raw.item["centrifuge"].stack_size,
    place_result = "blood-centrifuge",
  },
  {
    type = "item",
    name = "biter-spawner",
    icon = "__RichBlood__/graphics/icons/biter-spawner.png",
    icon_size = 64,
    subgroup = "blood-processing",
    order = "a[blood]-53",
    stack_size = 50,
    place_result = "biter-spawner",
  },
  {
    type = "item",
    name = "BRISPR-enhanced-biter-spawner",
    icon = "__RichBlood__/graphics/icons/BRISPR-enhanced-biter-spawner.png",
    icon_size = 64,
    subgroup = "blood-processing",
    order = "a[blood]-54",
    stack_size = 50,
    place_result = "BRISPR-enhanced-biter-spawner",
  },
})
